### PR TITLE
[stdlib] Add basics of a `collections.Counter`

### PR DESCRIPTION
**Pull request by msaelices**
Implement `collections.Counter` class. Not fully implemented YET.

There is a much more complete PR version here: https://github.com/modularml/mojo/pull/2910, but I was asked to split the PR into smaller ones, so this is the first one.

### CODE REVIEW

File: stdlib/src/collections/counter.mojo [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    #    Boolable
+):
+    """A container for counting hashable items.
+
+    The value type must be specified statically, unlike a Python
+    Counter, which can accept arbitrary value types.
+
+    The value type must implement the `KeyElement` trait, as its values are
+    stored in the dictionary as keys. `KeyElement` includes
+    `Movable`, `Hashable`, and `EqualityComparable`.
```
Comment (laszlokindrat) [inline stdlib/src/collections/counter.mojo:29]:
```suggestion
    stored in the dictionary as keys.
```

File: stdlib/src/collections/counter.mojo [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    def __init__(inout self, items: List[V]):
+        """Create a from an input iterable.
+
+        Args:
+            items: A list of items to count.
+
+        Usage:
+        ```mojo
+        c = Counter(List('g', 'a', 't', 't', 'a', 'c', 'a'))
+        ```
```
Comment (laszlokindrat) [inline stdlib/src/collections/counter.mojo:66]:
```suggestion
```

File: stdlib/src/collections/counter.mojo [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    var _data: Dict[V, Int]
+
+    def __init__(inout self):
+        """Create a new, empty Counter object.
+
+        Usage:
+        ```mojo
+        c = Counter()
+        ```
```
Comment (laszlokindrat) [inline stdlib/src/collections/counter.mojo:52]: Here and below: these are trivial examples, so they are not needed.
```suggestion
```

**Review by laszlokindrat — APPROVED** [review]
Great, thanks! That was really easy to review, and I only have a couple quibbles.

**Comment by laszlokindrat** [comment]
Looks like it's failing the formatter?

**Comment by msaelices** [comment]
> Looks like it's failing the formatter?

Fixed here: https://github.com/msaelices/mojo/commit/5be048bf0020598a02146e89b9f24665120eb27e

**Comment by laszlokindrat** [comment]
!sync

**Pull request closed without merging**